src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionXpathSingleFilter.java

```diff
@@ -0,0 +1,63 @@
+package com.puppycrawl.tools.checkstyle.filters;
+
+import com.puppycrawl.tools.checkstyle.TreeWalkerAuditEvent;
+import com.puppycrawl.tools.checkstyle.TreeWalkerFilter;
+import com.puppycrawl.tools.checkstyle.api.AutomaticBean;
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+
+/**
+ * 
+ */
+public class SuppressionXpathSingleFilter extends AutomaticBean implements
+        TreeWalkerFilter {
+    private XpathFilter xpathFilter;
+
+    /** The pattern for file names. */
+    private String filePattern;
+
+    /** The pattern for check class names. */
+    private String checkPattern;
+
+    /** The pattern for message names. */
+    private String messagePattern;
+
+    /** Module id filter. */
+    private String moduleId;
+
+    /** Xpath query. */
+    private String xpathQuery;
+
+    public void setFilePattern(String filePattern) {
+        this.filePattern = filePattern;
+    }
+
+    public void setCheckPattern(String checkPattern) {
+        this.checkPattern = checkPattern;
+    }
+
+    public void setMessagePattern(String messagePattern) {
+        this.messagePattern = messagePattern;
+    }
+
+    public void setModuleId(String moduleId) {
+        this.moduleId = moduleId;
+    }
+
+    public void setXpathQuery(String xpathQuery) {
+        this.xpathQuery = xpathQuery;
+    }
+
+    public SuppressionXpathSingleFilter() {
+        xpathFilter = new XpathFilter(filePattern, checkPattern, messagePattern, moduleId, xpathQuery);
```

Comment:
This will not work as setters will be called after constructor. This type of line will need to be in the `accept` method.